code/headers/nmea_listener.hpp

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include <usart_connection.hpp>
+
+namespace r2d2::location {
+    class nmea_listener_c {
+    protected:
+        constexpr static uint8_t start_byte = '$';
+        constexpr static uint8_t end_byte = '\n';
+        constexpr static size_t buffer_size = 100;
+
+        usart::usart_connection_c &usart;
+
+        /**
+         * @brief buffer for storing the nmea string
+         *
+         */
+        uint8_t buffer[buffer_size];
+
+        /**
+         * @brief index where the next location of data should be
+         *
+         */
+        size_t index = 0;
+
+        /**
+         * @brief bool for storing if a string is avaialable
+         *
+         */
+        bool available;
+
+    public:
+        nmea_listener_c(usart::usart_connection_c &usart) : usart(usart) {
+        }
+
+        bool nmea_available() {
```

Comment:
This function, along with some others may be `const`

Reply:
made some of them constexpr